Documentation/Proposals/NNNN-exit-tests.md

```diff
@@ -0,0 +1,787 @@
+the testing library:
+
+```swift
+/// Check that an expression causes the process to terminate in a given fashion.
+///
+/// - Parameters:
+///   - expectedExitCondition: The expected exit condition.
+///   - observedValues: An array of key paths representing results from within
+///     the exit test that should be observed and returned by this macro. The
+///     ``ExitTest/Result/statusAtExit`` property is always returned.
+///   - comment: A comment describing the expectation.
+///   - sourceLocation: The source location to which recorded expectations and
+///     issues should be attributed.
+///   - expression: The expression to be evaluated.
+///
+/// - Returns: If the exit test passed, an instance of ``ExitTest/Result``
+///   describing the state of the exit test when it exited. If the exit test
+///   fails, the result is `nil`.
+///
+/// Use this overload of `#expect()` when an expression will cause the current
+/// process to terminate and the nature of that termination will determine if
+/// the test passes or fails. For example, to test that calling `fatalError()`
+/// causes a process to terminate:
+///
+/// await #expect(exitsWith: .failure) {
+///   fatalError()
+/// }
+///
+/// - Note: A call to this expectation macro is called an "exit test."
+///
+/// ## How exit tests are run
+///
+/// When an exit test is performed at runtime, the testing library starts a new
+/// process with the same executable as the current process. The current task is
+/// then suspended (as with `await`) and waits for the child process to
+/// terminate. `expression` is not called in the parent process.
+///
+/// Meanwhile, in the child process, `expression` is called directly. To ensure
+/// a clean environment for execution, it is not called within the context of
+/// the original test. If `expression` does not terminate the child process, the
+/// process is terminated automatically as if the main function of the child
+/// process were allowed to return naturally. If an error is thrown from
+/// `expression`, it is handed as if the error were thrown from `main()` and the
+/// process is terminated.
+///
+/// Once the child process terminates, the parent process resumes and compares
+/// its exit status against `expectedExitCondition`. If they match, the exit
+/// test has passed; otherwise, it has failed and an issue is recorded.
+///
+/// ## Child process output
+///
+/// By default, the child process is configured without a standard output or
+/// standard error stream. If your test needs to review the content of either of
+/// these streams, you can pass its key path in the `observedValues` argument:
+///
+/// let result = await #expect(
+///   exitsWith: .failure,
+///   observing: [\.standardOutputContent]
+/// ) {
+///   print("Goodbye, world!")
+///   fatalError()
+/// }
+/// if let result {
+///   #expect(result.standardOutputContent.contains(UInt8(ascii: "G")))
+/// }
```

Comment:
Either indent or use backticks around this code listing, otherwise it will render as a paragraph.

Reply:
This is a code block inside a code block. It doesn't need to be rendered.